sqrl-server/sqrl-server-core/src/main/java/com/datasqrl/graphql/server/GraphQLEngineBuilder.java

```diff
@@ -70,12 +71,35 @@ public class GraphQLEngineBuilder implements
           .build())
       .build();
 
-  public GraphQLEngineBuilder() {
-    addlTypes = new ArrayList<>();
+  private GraphQLEngineBuilder(Builder builder) {
+    this.addlTypes = builder.addlTypes;
+    this.subscriptionConfiguration = builder.subscriptionConfiguration;
+    this.mutationConfiguration = builder.mutationConfiguration;
   }
 
-  public GraphQLEngineBuilder(List<GraphQLScalarType> addlTypes) {
-    this.addlTypes = addlTypes;
+  public static class Builder {
+    private List<GraphQLScalarType> addlTypes = new ArrayList<>();
+    private SubscriptionConfiguration<DataFetcher<?>> subscriptionConfiguration;
+    private MutationConfiguration<DataFetcher<?>> mutationConfiguration;
+
+    public Builder withAdditionalTypes(List<GraphQLScalarType> types) {
+      this.addlTypes = types;
+      return this;
+    }
+
+    public Builder withSubscriptionConfiguration(SubscriptionConfiguration<DataFetcher<?>> configurer) {
+      this.subscriptionConfiguration = configurer;
+      return this;
+    }
+
+    public Builder withMutationConfiguration(MutationConfiguration<DataFetcher<?>> configurer) {
+      this.mutationConfiguration = configurer;
+      return this;
+    }
+
+    public GraphQLEngineBuilder build() {
+      return new GraphQLEngineBuilder(this);
+    }
   }
```

Comment:
This pattern doesn't conform to the current architecture, let's choose one way to do data fetchers. The current pattern is the `context` produces the data fetchers. We could move to a typed GraphQLEngineBuilder for both servlet types and move it out of the context, but we shouldn't have a quasi pattern where we construct query fetchers one way and subscriptions/mutation fetchers a different way.